### ISSUE TITLE
permission->assignRole after Permission::findOrCreate is not correctly assigning the roles

### ISSUE THREAD

**Issue by tiagocpeixoto** [spatie/laravel-permission#2757]
### Description

When I run the database seeder for roles and permissions, permissons' ``assignRole`` is not correctly adding new roles for the same permission that is queried from ``Permission::findOrCreate``. 

With the example application I provided below, only the first role is assigned.

I noticed this in version 6.10. When I revert to version 6.9, the roles are assigned as expected.

### Steps To Reproduce

1. Create at least two roles (``$role1`` and ``$role2``);
2. Create the permission using ``Permission::findOrCreate('Permission 1')``; 
3. Assign the first role to the permission using ``$permission->assignRole($role1)``;
4. Get the created permission using ``Permission::findOrCreate('Permission 1')``; 
5. Then assign the second role to the same permission also using ``$permission->assignRole($role2)``

### Example Application

```
$role1 = Role::create(['name' => 'Role 1']);
$role2 = Role::create(['name' => 'Role 2']);

$permission = Permission::findOrCreate('Permission 1');
$permission->assignRole($role1);

$permission = Permission::findOrCreate('Permission 1');
$permission->assignRole($role2);
```

### Version of spatie/laravel-permission package:

6.10

### Version of laravel/framework package:

11

### PHP version:

8.2

### Database engine and version:

SQLite

### OS: Windows/Mac/Linux version:

Windows 11 and MacOS 15

**Comment by drbyte** [comment]
Does the fix in `6.10.1` resolve this problem?

**Comment by tiagocpeixoto** [comment]
> Does the fix in `6.10.1` resolve this problem?

Yes, thanks!

**Comment by drbyte** [comment]
Closed by #2759
